src/env/tock/mod.rs

```diff
@@ -12,18 +14,48 @@ use libtock_drivers::console::Console;
 use libtock_drivers::result::{FlexUnwrap, TockError};
 use libtock_drivers::timer::Duration;
 use libtock_drivers::{led, timer, usb_ctap_hid};
+use persistent_store::StorageResult;
+
+mod storage;
 
 pub struct TockEnv {
     rng: TockRng256,
+    storage: bool,
+    upgrade_storage: bool,
 }
 
 impl TockEnv {
-    pub fn new() -> Self {
-        let rng = TockRng256 {};
-        TockEnv { rng }
+    /// Returns the unique instance of the Tock environment.
+    ///
+    /// This function returns `Some` the first time it is called. Afterwards, it repeatedly returns
+    /// `None`.
+    pub fn new() -> Option<Self> {
```

Comment:
Good question. Indeed it's probably better to panic here. In particular because I foresee that we may panic for other reasons during setup in this function.